bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java

```diff
@@ -84,6 +88,21 @@
      */
     String getCallToActionText();
 
+    /**
+     * Returns the call to action command for the configured call to action.
+     *
+     * @return "addToCart" or "details" or {@code null} if call to action is not configured
+     */
+    default String getCallToActionCommand() {
```

Comment:
A new public method is unnecessary to resolve this neither is a change to the htl necessary. You can achieve the same outcome without touching the public interface. 

See the following implementation only proposal
```@Override
public String getCallToAction() {
    if (StringUtils.isNotEmpty(cta)) {
        // return always details if the product cannot be added to cart (e.g. configurable product)
        return oneClickShoppable(getProduct()) ? cta : "details";
    }
    return null;
}

@Override
public String getCallToActionText() {
    if (StringUtils.isNotEmpty(ctaText)) {
        return ctaText;
    } else if ("details".equals(cta)) {
        return getI18n().get("See more details");
    } else if ("add-to-cart".equals(cta)) {
        return getI18n().get("Add to Cart");
    }
    return null;
}
```

Reply:
This seems to fix the issue with minimal impact. The logic behind it is quite tricky that one may not want to multiply in the future but may work for the current case.

Reply:
If we have other cases in the future, it makes sense to revisit it but now we dont. As discussed, doing a minimal impact change for a bug that will effectively address the issue for all customers (no matter of `actions.html` is overwritten or not) seems reasonable reasonable and more favourable.   

If it is _tricky_ or not is a subjective opinion. The "command" being not-equal to the "action" is tricky to understand either + it requires more logic on the client to figure out what to finally do.